File: test_project/gen_code_java/tbbag.java
package demo;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class TbBag {
    private final Map<int, Bag> _dataMap;
    private final List<Bag> _dataList;

    public TbBag(List<Bag> dataList) {
        this._dataList = dataList;
        this._dataMap = new HashMap<>();
        for (Bag v : _dataList) {
            this._dataMap.put(v.Id, v);
        }
    }

    public Map<int, Bag> getDataMap() {
        return _dataMap;
    }

    public List<Bag> getDataList() {
        return _dataList;
    }

    public Bag get(int key) {
        return _dataMap.get(key);
    }
}

File: test_project/gen_code_lua/tbbag.lua
-- Table TbBag
local TbBag = {
    _dataMap = {},
    _dataList = {}
}

function TbBag.Load(dataList)
    TbBag._dataList = dataList
    for _, v in ipairs(dataList) do
        TbBag._dataMap[v.Id] = v
    end
end

function TbBag.Get(key)
    return TbBag._dataMap[key]
end

function TbBag.GetDataList()
    return TbBag._dataList
end

function TbBag.GetDataMap()
    return TbBag._dataMap
end

return TbBag